src/mock.rs

```diff
@@ -0,0 +1,28 @@
+use crate::expectation::model::{Expectation, HttpMethod};
+
+pub struct IncomingRequest {
+    pub method: HttpMethod,
+    pub code: i32,
+    pub path: String,
+}
+
+pub fn look_for_expectation(
+    expectations: Vec<&Expectation>,
+    req: IncomingRequest,
+) -> Option<&Expectation> {
+    for e in expectations {
```

Comment:
Working on a `iter()` with a [`find`](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.find) would maybe a better struct to avoid return in a `loop`. Could also provide a way to define predicates outside of the loop search.